cli/azd/pkg/project/service_target_ml_endpoint.go

```diff
@@ -0,0 +1,282 @@
+		func(ctx context.Context, args ProjectLifecycleEventArgs) error {
+			projectName := m.env.Getenv(AiProjectNameEnvVarName)
+			aiStudioLink := ai.AzureAiStudioLink(
+				m.env.GetTenantId(),
+				m.env.GetSubscriptionId(),
+				m.env.Getenv(environment.ResourceGroupEnvVarName),
+				projectName,
+			)
+
+			err := m.env.Config.Set("provision.links.aiStudio", &output.Link{
+				Name:        "Azure AI Studio",
+				Description: fmt.Sprintf("View the %s project in Azure AI studio:", projectName),
+				Url:         aiStudioLink,
+			})
+			if err != nil {
+				return fmt.Errorf("failed setting aiStudio link: %w", err)
+			}
+
+			if err := m.envManager.Save(ctx, m.env); err != nil {
+				return fmt.Errorf("failed saving environment: %w", err)
+			}
+
+			return nil
+		},
+	)
+}
+
+// RequiredExternalTools returns the required external tools for the machineLearningEndpointTarget
+func (m *machineLearningEndpointTarget) RequiredExternalTools(ctx context.Context) []tools.ExternalTool {
+	return m.aiHelper.RequiredExternalTools(ctx)
+}
+
+// Package packages the service for deployment to an Azure ML online endpoint
+// This method is a no-op since the actual packaging is handled by the underlying AI/ML Python SDKs
+func (m *machineLearningEndpointTarget) Package(
+	ctx context.Context,
+	serviceConfig *ServiceConfig,
+	frameworkPackageOutput *ServicePackageResult,
+) *async.TaskWithProgress[*ServicePackageResult, ServiceProgress] {
+	// Implement the Package method here.
+	return async.RunTaskWithProgress(func(task *async.TaskContextWithProgress[*ServicePackageResult, ServiceProgress]) {
+		task.SetResult(&ServicePackageResult{})
+	})
+}
+
+// Deploy deploys the service to an Azure ML online endpoint
+func (m *machineLearningEndpointTarget) Deploy(
+	ctx context.Context,
+	serviceConfig *ServiceConfig,
+	servicePackage *ServicePackageResult,
+	targetResource *environment.TargetResource,
+) *async.TaskWithProgress[*ServiceDeployResult, ServiceProgress] {
+	// Implement the Deploy method here.
+	return async.RunTaskWithProgress(func(task *async.TaskContextWithProgress[*ServiceDeployResult, ServiceProgress]) {
+		endpointConfig, err := ai.ParseConfig[ai.EndpointDeploymentConfig](serviceConfig.Config)
+		if err != nil {
+			task.SetError(err)
+			return
+		}
+
+		workspaceScope, err := m.getWorkspaceScope(serviceConfig, targetResource)
+		if err != nil {
+			task.SetError(err)
+			return
+		}
+
+		deployResult := &EndpointDeploymentResult{}
+
+		// Initialize the AI project that will be used for the python bridge
+		task.SetProgress(NewServiceProgress("Initializing AI project"))
+		if err := m.aiHelper.Initialize(ctx); err != nil {
+			task.SetError(fmt.Errorf("failed initializing AI project: %w", err))
+			return
+		}
+
+		// Ensure the workspace is valid
+		if err := m.aiHelper.EnsureWorkspace(ctx, workspaceScope); err != nil {
+			task.SetError(
+				fmt.Errorf("workspace '%s' was not found within subscription '%s' and resource group '%s': %w",
```

Comment:
Just checking my understanding - the way to fix this issue is to ensure you've created the resource in your bicep and that `AZUREML_AI_PROJECT_NAME` is set as an output?

Reply:
Yes you can either set the environment variable or you can set the `workspace` param in the `azure.yaml` for the service config